README.md

```diff
@@ -1 +1,103 @@
-# che-theia-factory-extension
\ No newline at end of file
+# Che Theia factory
+This Theia extension is performing additional actions while creating/starting a Che workspace through a factory URL.
+Provides basic factory client side clone feature
+
+- retrieving the factory-id from the URL
+- requesting Che factory API to get the factory definition
+
+
+From the factory definition:
+- cloning projects defined in the factory definition
+- checking out the right branch if needed
+
+## Testing this extension
+Having that
+
+- We have two or three Github projects to import in a factory,
+- One of the project has a specific branch to checkout.
+
+When:
+
+1. Create a new empty workspace (blank or any other type but that works with the Che legacy IDE).
+2. Start the workspace. Import projects from Che legacy IDE (we do that so the project types are defined, could also be done manually without having to start a workspace). 
+3. Checkout the specific branch for one of the project.
+3. Create a new factory from the created workspace.
+4. Edit the factory and select `from a stack` and `Java Theia (OpenShift)`
+5. Edit the configuration and add to the theia machine the environment variable THEIA_PLUGINS
```

Comment:
could you please use
THEIA_EXTENSIONS (as THEIA_PLUGINS will now only work with theia plugins)